Dockerfile

```diff
@@ -0,0 +1,26 @@
+# Base
+FROM python as base
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+WORKDIR /code
+
+ENV PATH "/code:/root:/root/.poetry/bin:${PATH}"
+
+# add and install python requirements
+COPY pyproject.toml poetry.lock /code/
+RUN poetry install
+
+# Copy the rest of the code
+COPY / /code/
+
+# Development 
+FROM base as development
+EXPOSE 5100
+CMD ["poetry", "run", "flask", "run", "--host", "0.0.0.0", "-p", "5100"]
```

Comment:
Note that there's no reason to make this use a different port from the production version